azurerm/resource_arm_cognitive_account.go

```diff
@@ -0,0 +1,258 @@
+			"location":            locationSchema(),
+			"tags":                tagsSchema(),
+			"resource_group_name": resourceGroupNameSchema(),
+		},
+	}
+}
+
+func resourceArmCognitiveAccountSkuHash(v interface{}) int {
+	var buf bytes.Buffer
+	m := v.(map[string]interface{})
+	buf.WriteString(fmt.Sprintf("%v-", m["name"]))
+	buf.WriteString(fmt.Sprintf("%v-", m["tier"]))
+
+	return hashcode.String(buf.String())
+}
+
+type dummy struct{}
+
+func resourceArmCognitiveAccountCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cognitiveAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+	fmt.Printf("[INFO] preparing arguments for Azure ARM  Cognitive Servies Account creation.\n")
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	kind := d.Get("kind").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	fmt.Printf("[DEBUG] name: %s location: %s resGroup: %s kind: %s \n", name, location, resGroup, cognitiveservices.Kind(kind))
+	sku := expandCognitiveSku(d)
+	prop := dummy{}
+
+	createParameter := cognitiveservices.AccountCreateParameters{
+		Kind:       cognitiveservices.Kind(kind),
+		Location:   &location,
+		Sku:        &sku,
+		Properties: &prop,
+		Tags:       *expandTags(tags),
+	}
+
+	account, err := client.Create(ctx, resGroup, name, createParameter)
+	if err != nil {
+		fmt.Printf("Error: %s \n", err.Error())
+		return err
+	}
+	if account.AccountProperties.ProvisioningState != cognitiveservices.Succeeded {
+		return fmt.Errorf("Error provisioning Account")
+	}
+
+	d.SetId(*account.ID)
+
+	return nil
+
+}
+
+func expandCognitiveSku(d *schema.ResourceData) cognitiveservices.Sku {
+	skus := d.Get("sku").(*schema.Set).List()
+	sku := skus[0].(map[string]interface{})
+	name := cognitiveservices.SkuName(sku["name"].(string))
+	tier := cognitiveservices.SkuTier(sku["tier"].(string))
+	fmt.Printf("[DEBUG] SKU Name: %s Tier: %s\n", name, tier)
+	return cognitiveservices.Sku{
+		Name: name,
+		Tier: tier,
+	}
+}
+func flattenCognitiveSku(sku cognitiveservices.Sku) *schema.Set {
+	skus := &schema.Set{
+		F: resourceArmCognitiveAccountSkuHash,
+	}
+
+	defaultSku := make(map[string]interface{}, 2)
+
+	defaultSku["name"] = sku.Name
+	defaultSku["tier"] = sku.Tier
+	fmt.Printf("[DEBUG] flatten name: %s tier: %s", sku.Name, sku.Tier)
+
+	skus.Add(defaultSku)
+
+	return skus
+}
+
+func resourceArmCognitiveAccountRead(d *schema.ResourceData, meta interface{}) error {
+	fmt.Printf("[INFO] resource Arm Cognitive account Read.\n")
+
+	client := meta.(*ArmClient).cognitiveAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["accounts"]
+	resp, err := client.GetProperties(ctx, resGroup, name)
+	if err != nil {
+		return err
+	}
+	d.Set("name", resp.Name)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
```

Comment:
for older resources the Azure API doesn't always match the response schema, can we make this:

```
if location := resp.Location; location != nil {
d.Set("location", azureRMNormalizeLocation(*location))
}
```

to avoid a crash in these circumstances?